c7n/filters/related.py

```diff
@@ -49,6 +49,8 @@ def get_related_ids(self, resources):
             "[].%s" % self.RelatedIdsExpression, resources))
 
     def get_related(self, resources):
+        if hasattr(self, 'initialize'):
```

Comment:
this doesn't really feel like the right fix, you'll get duplicate initializations, its implicit, etc. its really up to the caller to set this up.

Reply:
Yeah, I'm not super happy with this either. Unfortunately, initializing in `get_related_ids` in the security-group filter will execute initialize on each resource, causing a flood of api calls to the launch configuration api. This looked like the originator of the call in the first place, but I could be wrong.